oligo_designer_toolsuite/database/_oligos_database.py

```diff
@@ -1,444 +1,465 @@
+############################################
+# imports
+############################################
+
 import os
 import warnings
+import pandas as pd
+
 from pathlib import Path
+from copy import copy, deepcopy
+from joblib import cpu_count, Parallel, delayed
+
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
 
-import pyfaidx
-from joblib import cpu_count
+from ..utils._data_parser import (
+    check_fasta_format,
+    check_tsv_format,
+    parse_fasta_header,
+)
 
-from ..utils import FtpLoaderEnsembl, FTPLoaderNCBI, _data_parser
-from ._oligos_generator import OligosGenerator
-from ._transcript_generator import TranscriptGenerator
+############################################
+# Oligo Database Class
+############################################
 
 
-class CustomOligoDB:
-    """This class generates all possible guides that can be designed for a given list of genes,
-    based on the transcriptome annotation or the gene CDS or the whole genome.
-    Moreover, it can write to a file and read the database containing all the oligos sequences.
-    The gtf and a fasta files are passed in input.
+class OligoDatabase:
+    """This class generates all possible oligos that can be designed for a given list of regions (e.g. genes),
+    based on the transcriptome or the gene CDS annotation or the whole genome provided as fasta file.
+    Moreover, the database can be saved and loaded to/from a tsv file.
```

Comment:
Did we remove the option to write a gtf file?

Reply:
yes, I removed the option for the gtf file, but added an option to just write the sequences with IDs to a fasta file. I think that's sufficient in the end. The tsv reader/writer is now the save/loader for the database